### PR TITLE
[WIP] Update Open Street Map examples to use spatialpandas

### PR DESCRIPTION
**Pull request by jonmmease**
WIP PR to update the Open Street Map examples (1 billion and 3 billion) to use `spatialpandas` rather than the legacy `datashader.spatial` module:

TODO:
 - [ ] Upload the two new parquet datasets to the `datashader-data` s3 bucket. @jlstevens can you give me access?  Note, I'm planning to give the new files a different name so that the old examples tutorial will still work. (New names osm-1b.parq and osm-3b.parq.)
 - [ ] `DaskGeoDataFrame` support in HoloViews is still needed to  make the interactive HoloViews examples work. See https://github.com/holoviz/holoviews/issues/4234.
 - [ ] Update census example as well: https://examples.pyviz.org/census/census.html.

### CODE REVIEW

**Comment by jbednar** [comment]
Thanks; I've requested S3 access for you.  For the new names, I suggest osm-1b.parq and osm-3b.parq.  What about the Census dataset?

File: osm/osm-3billion.ipynb [hunk 1]
```diff
@@ -10,7 +10,7 @@
...
+    "The data is taken from Open Street Map's (OSM) [bulk GPS point data](https://blog.openstreetmap.org/2012/04/01/bulk-gps-point-data/), and is unfortunately too large to distribute with Datashader (8.4GB compressed). The data was collected by OSM contributors' GPS devices, and was provided as a CSV file of `latitude,longitude` coordinates. The data was downloaded from their website, extracted, converted to use positions in Web Mercator format using `datashader.utils.lnglat_to_meters()`, sorted using [spacial indexing](https://nbviewer.jupyter.org/github/holoviz/spatialpandas/blob/master/examples/Overview.ipynb#Pack-partitions), and then stored in a [parquet](https://arrow.apache.org/docs/python/parquet.html) file for faster disk access. See the [spatialpandas](https://github.com/holoviz/spatialpandas) project for more information. To run this notebook, you would need to do the same process yourself to obtain `osm-3billion.parq`.   Once you have it, you can follow the steps below to load and plot the data."
```
Comment (jbednar) [inline osm/osm-3billion.ipynb]:
```suggestion
    "The data is taken from Open Street Map's (OSM) [bulk GPS point data](https://blog.openstreetmap.org/2012/04/01/bulk-gps-point-data/), and is unfortunately too large to distribute with Datashader (8.4GB compressed). The data was collected by OSM contributors' GPS devices, and was provided as a CSV file of `latitude,longitude` coordinates. The data was downloaded from their website, extracted, converted to use positions in Web Mercator format using `datashader.utils.lnglat_to_meters()`, sorted using [spatial indexing](https://nbviewer.jupyter.org/github/holoviz/spatialpandas/blob/master/examples/Overview.ipynb#Pack-partitions), and then stored in a [parquet](https://arrow.apache.org/docs/python/parquet.html) file for faster disk access. See the [spatialpandas](https://github.com/holoviz/spatialpandas) project for more information. To run this notebook, you would need to do the same process yourself to obtain `osm-3billion.parq`.   Once you have it, you can follow the steps below to load and plot the data."
```

File: osm/osm-1billion.ipynb [hunk 2]
```diff
@@ -8,14 +8,14 @@
...
+    "The data is taken from Open Street Map's (OSM) [bulk GPS point data](https://blog.openstreetmap.org/2012/04/01/bulk-gps-point-data/). The data was collected by OSM contributors' GPS devices, and was provided as a CSV file of `latitude,longitude` coordinates. The data was downloaded from their website, extracted, converted to use positions in Web Mercator format using `datashader.utils.lnglat_to_meters()`, sorted using [spacial indexing](https://nbviewer.jupyter.org/github/holoviz/spatialpandas/blob/master/examples/Overview.ipynb#Pack-partitions), and then stored in a [parquet](https://arrow.apache.org/docs/python/parquet.html) file for faster disk access. See the [spatialpandas](https://github.com/holoviz/spatialpandas) project for more information. As usual, loading it into memory will be the most time-consuming step, by far:\n",
```
Comment (jbednar) [inline osm/osm-1billion.ipynb]:
```suggestion
    "The data is taken from Open Street Map's (OSM) [bulk GPS point data](https://blog.openstreetmap.org/2012/04/01/bulk-gps-point-data/). The data was collected by OSM contributors' GPS devices, and was provided as a CSV file of `latitude,longitude` coordinates. The data was downloaded from their website, extracted, converted to use positions in Web Mercator format using `datashader.utils.lnglat_to_meters()`, sorted using [spatial indexing](https://nbviewer.jupyter.org/github/holoviz/spatialpandas/blob/master/examples/Overview.ipynb#Pack-partitions), and then stored in a [parquet](https://arrow.apache.org/docs/python/parquet.html) file for faster disk access. See the [spatialpandas](https://github.com/holoviz/spatialpandas) project for more information. As usual, loading it into memory will be the most time-consuming step, by far:\n",
```

**Comment by jonmmease** [comment]
> For the new names, I suggest osm-1b.parq and osm-3b.parq.

Sounds good

> What about the Census dataset?

Ah, right. I forgot that one had a `datashader.spatial` dataset. I added another TODO above.

**Comment by maximlt** [comment]
It looks like part of the changes made in this PR were also made in this [mega PR](https://github.com/pyviz-topics/examples/pull/167/files#diff-88e40c73c0adb1e9e020883a271d09503bf481790a69b46ceb0731cb4d0eed88), hence the conflicts. It seems though that some more changes were planned in this PR, they'll need a closer look.

Pinging @ianthomas23 as this example ([OSM 1B](https://examples.pyviz.org/osm/osm-1billion.html#osm-gallery-osm-1billion) and [OSM 3B](https://examples.pyviz.org/osm/osm-3billion.html#osm-gallery-osm-3billion)) use datashader and spatialpandas, looking at them you may have ideas for potential improvements.

**Comment by jonmmease** [comment]
Closing to clean up my PR history, but feel free to re-open if someone wants to pick this up again.

**Pull request closed without merging**